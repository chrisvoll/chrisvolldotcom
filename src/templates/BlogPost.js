import React from 'react';
import Helmet from 'react-helmet';
import WithFadeTransition from '../components/WithFadeTransition';
import { StaticQuery, graphql } from 'gatsby';

import { HeaderGroup, Tag, Headline, Date } from '../util/style';

import 'prismjs/themes/prism-solarizedlight.css';

export default function BlogPost() {
  return (
    <StaticQuery
      query={graphql`
        query BlogPostByPath($path: String!) {
          markdownRemark(frontmatter: { path: { eq: $path } }) {
            html
            excerpt(pruneLength: 160)
            frontmatter {
              date(formatString: "MMMM DD, YYYY")
              path
              title
              tags
            }
          }
        }
      `}
      render={data => {
        const post = data && data.markdownRemark;
        const category = post.frontmatter.tags[0];
        const meta = [
          {
            name: 'description',
            content: post.excerpt
          },
          {
            name: 'keywords',
            content: post.frontmatter.tags.join(', ')
          }
        ];
        return (
          <WithFadeTransition>
            <Helmet title={post.frontmatter.title} meta={meta} />
            <HeaderGroup>
              <Tag to={`/tags/${category}`}>{category}</Tag>
              <Headline>{post.frontmatter.title}</Headline>
              <Date>{post.frontmatter.date}</Date>
            </HeaderGroup>
            <div
              className="post-content"
              dangerouslySetInnerHTML={{ __html: post.html }}
            />
          </WithFadeTransition>
        );
      }}
    />
  );
}
