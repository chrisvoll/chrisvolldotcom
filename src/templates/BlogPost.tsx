import { graphql } from 'gatsby';
import React from 'react';
import Helmet from 'react-helmet';
import WithFadeTransition from '../components/WithFadeTransition';
import Layout from '../layouts/layout';
import { IPost } from '../types';
import { Date, HeaderGroup, Headline, Tag } from '../util/style';

import 'prismjs/themes/prism-solarizedlight.css';

interface IProps {
  data: {
    markdownRemark: IPost;
  };
}

const BlogPost: React.FC<IProps> = ({ data }) => {
  const post = data && data.markdownRemark;
  const category = post.frontmatter.tags[0];
  const meta = [
    {
      name: 'description',
      content: post.excerpt
    },
    {
      name: 'keywords',
      content: post.frontmatter.tags.join(', ')
    }
  ];
  return (
    <Layout>
      <WithFadeTransition>
        <Helmet title={post.frontmatter.title} meta={meta} />
        <HeaderGroup>
          <Tag to={`/tags/${category}`}>{category}</Tag>
          <Headline>{post.frontmatter.title}</Headline>
          <Date>{post.frontmatter.date}</Date>
        </HeaderGroup>
        <div
          className="post-content"
          dangerouslySetInnerHTML={{ __html: post.html }}
        />
      </WithFadeTransition>
    </Layout>
  );
};

export default BlogPost;

export const pageQuery = graphql`
  query BlogPostByPath($path: String!) {
    markdownRemark(frontmatter: { path: { eq: $path } }) {
      html
      excerpt(pruneLength: 160)
      frontmatter {
        date(formatString: "MMMM DD, YYYY")
        path
        title
        tags
      }
    }
  }
`;
