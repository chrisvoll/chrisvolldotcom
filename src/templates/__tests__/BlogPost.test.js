import React from 'react';
import BlogPost from '../BlogPost';

describe('Blog Post', () => {
  let data;

  beforeEach(() => {
    data = {
      markdownRemark: {
        frontmatter: {
          title: 'Sample Title',
          path: '/sample-title',
          date: 'September 14, 2009'
        },
        html: '<p><strong>html</strong> output</p>'
      }
    };
  });

  function renderShallow() {
    return shallow(<BlogPost data={data} />);
  }

  it('can render a default post', () => {
    expect(renderShallow()).toMatchSnapshot();
  });
});
