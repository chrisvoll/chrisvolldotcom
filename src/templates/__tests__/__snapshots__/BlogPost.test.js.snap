// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Post can render a default post 1`] = `
<WithFadeTransition
  delay={0}
>
  <div>
    <HelmetWrapper
      encodeSpecialCharacters={true}
      title="Bloggyblog - Sample Title"
    />
    <styled.div>
      <Styled(GatsbyLink)
        to="/tags/tag-a"
      >
        tag-a
      </Styled(GatsbyLink)>
      <styled.h1>
        Sample Title
      </styled.h1>
      <styled.h3>
        September 14, 2009
      </styled.h3>
    </styled.div>
    <p>
      Tags:
       
      <GatsbyLink
        to="/tags/tag-b"
      >
        tag-b
      </GatsbyLink>
    </p>
    <div
      className="post-content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p><strong>html</strong> output</p>",
        }
      }
    />
  </div>
</WithFadeTransition>
`;
